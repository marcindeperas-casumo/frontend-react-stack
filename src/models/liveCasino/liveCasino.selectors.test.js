// @flow
import {
  liveTableEntitySelector,
  liveTableSelector,
  groupLiveGames,
  mapLobbyTranslations,
} from "./liveCasino.selectors";

describe("Models/Jackpots/Selectors", () => {
  const state = {
    schema: {
      liveTable: {
        one: { tableId: "one" },
        two: { tableId: "two" },
        three: { tableId: "three" },
        four: { tableId: "four" },
      },
    },
  };

  describe("liveTableEntitySelector()", () => {
    test("selects liveTables from state", () => {
      expect(liveTableEntitySelector(state)).toEqual(state.schema.liveTable);
    });

    test("returns empty object if no liveTables yet", () => {
      expect(liveTableEntitySelector({})).toEqual({});
    });
  });

  describe("liveTableSelector()", () => {
    test("returns liveTable object by id", () => {
      const id = "one";
      const table = liveTableSelector(id)(state);
      const expectedTable = state.schema.liveTable[id];

      expect(table).toEqual(expectedTable);
    });

    test("returns 'null' if liveTable doesn not exist", () => {
      const id = "one";
      const liveTable = liveTableSelector(id)({});

      expect(liveTable).toEqual(null);
    });
  });

  describe("mapLobbyTranslations()", () => {
    const input = [
      { key: "ultimate_texas_holdem", value: "xyz" },
      { key: "roulette", value: "asdf" },
      { key: "non_existing", value: "yolo" },
    ];
    const out = {
      UTH: input[0].value,
      Roulette: input[1].value,
      non_existing: input[2].value,
    };
    expect(mapLobbyTranslations(input)).toEqual(out);
  });

  describe("groupLiveGames()", () => {
    const liveGames1 = [
      {
        slug: "one",
        lobby: {
          type: "BlackJack",
          bets: {
            min: 0.2,
            max: 300,
          },
        },
      },
      {
        slug: "two",
        lobby: {
          type: "BlackJack",
          bets: {
            min: 0.1,
            max: 666,
          },
        },
      },
    ];
    const expected1 = [["BlackJack", ["two", "one"]]]; // order changes due to difference in bets range
    expect(groupLiveGames(liveGames1)).toEqual(expected1);

    const liveGames2 = [
      {
        slug: "one",
        lobby: {
          type: "BlackJack",
          bets: {
            min: 0.1,
            max: 1000,
          },
        },
      },
      {
        slug: "two",
        lobby: {
          type: "BlackJack",
          bets: {
            min: 0.1,
            max: 666,
          },
        },
      },
      {
        slug: "three",
        lobby: {
          type: "MoneyWheel",
          bets: {
            min: 0.1,
            max: 1000,
          },
        },
      },
    ];
    const expected2 = [
      ["BlackJack", ["one", "two"]],
      ["MoneyWheel", ["three"]],
    ]; // order changes due to difference in bets range
    expect(groupLiveGames(liveGames2)).toEqual(expected2);
  });
});
