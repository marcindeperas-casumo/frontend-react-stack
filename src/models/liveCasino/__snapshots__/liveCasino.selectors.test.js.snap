// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  Object {
    "disabledForLoggedOut": false,
    "hasPlayForFun": false,
    "inMaintenanceMode": false,
    "jackpotId": null,
    "lobby": Object {
      "betBehind": null,
      "bets": Object {
        "max": 2500,
        "min": 0.1,
        "symbol": "€",
      },
      "image": "https://lob.egcvi.com/thumbnail/csor1_mw_med_L.jpg",
      "players": 2,
      "provider": "casumo",
      "results": Array [
        "01",
        "01",
        "05",
        "01",
        "01",
        "01",
        "02",
        "01",
        "02",
      ],
      "seats": null,
      "tableId": "munbzmuueehqaavs",
      "type": "MoneyWheel",
    },
    "logo": "https://cms.casumo.com/wp-content/uploads/2018/08/TTLogo.png",
    "logoBackground": "https://cms.casumo.com/wp-content/uploads/2018/08/TopwheelTreasuresBack.png",
    "name": "Topwheel Treasures",
    "slug": "topwheel-treasures",
    "tableId": "munbzmuueehqaavs",
  },
  Object {
    "disabledForLoggedOut": false,
    "hasPlayForFun": false,
    "inMaintenanceMode": false,
    "jackpotId": null,
    "lobby": Object {
      "betBehind": null,
      "bets": Object {
        "max": 10000,
        "min": 1,
        "symbol": "€",
      },
      "history": Array [
        "P",
        "B",
        "P",
        "T",
        "B",
        "B",
        "P",
        "P",
      ],
      "image": "https://lob.egcvi.com/thumbnail/bac2_bs_med_L.jpg",
      "players": 1,
      "provider": "evolution",
      "results": Array [
        "P",
        "B",
        "P",
        "B",
        "B",
        "B",
        "B",
        "B",
        "B",
        "B",
        "B",
        "P",
        "P",
        "B",
        "P",
      ],
      "road": Array [
        Object {
          "bankerPair": false,
          "color": "Blue",
          "location": Object {
            "column": 0,
            "row": 0,
          },
          "natural": false,
          "playerPair": false,
          "score": 7,
          "ties": 0,
        },
        Object {
          "bankerPair": false,
          "color": "Blue",
          "location": Object {
            "column": 0,
            "row": 1,
          },
          "natural": false,
          "playerPair": false,
          "score": 6,
          "ties": 0,
        },
        Object {
          "bankerPair": false,
          "color": "Red",
          "location": Object {
            "column": 1,
            "row": 0,
          },
          "natural": true,
          "playerPair": false,
          "score": 8,
          "ties": 0,
        },
        Object {
          "bankerPair": true,
          "color": "Red",
          "location": Object {
            "column": 1,
            "row": 1,
          },
          "natural": false,
          "playerPair": false,
          "score": 9,
          "ties": 1,
        },
        Object {
          "bankerPair": false,
          "color": "Blue",
          "location": Object {
            "column": 2,
            "row": 0,
          },
          "natural": false,
          "playerPair": false,
          "score": 2,
          "ties": 0,
        },
        Object {
          "bankerPair": false,
          "color": "Red",
          "location": Object {
            "column": 3,
            "row": 0,
          },
          "natural": false,
          "playerPair": false,
          "score": 7,
          "ties": 0,
        },
        Object {
          "bankerPair": false,
          "color": "Blue",
          "location": Object {
            "column": 4,
            "row": 0,
          },
          "natural": false,
          "playerPair": false,
          "score": 4,
          "ties": 0,
        },
      ],
      "seats": null,
      "tableId": "oytmvb9m1zysmc44",
      "type": "Baccarat",
    },
    "logo": "https://cms.casumo.com/wp-content/uploads/2016/05/live-baccarat-logo.png",
    "logoBackground": "https://cms.casumo.com/wp-content/uploads/2016/05/live-baccarat-bg.jpg",
    "name": "Baccarat A",
    "slug": "baccarat-a",
    "tableId": "oytmvb9m1zysmc44",
  },
  Object {
    "disabledForLoggedOut": false,
    "hasPlayForFun": false,
    "inMaintenanceMode": false,
    "jackpotId": null,
    "lobby": Object {
      "betBehind": null,
      "bets": Object {
        "max": 15000,
        "min": 1,
        "symbol": "€",
      },
      "history": Array [
        "L",
        "R",
        "L",
        "T",
        "L",
        "T",
        "R",
        "L",
        "R",
        "R",
        "R",
      ],
      "image": "https://lob.egcvi.com/thumbnail/topcr1_bs_med_L.jpg",
      "players": 3,
      "provider": "evolution",
      "results": Array [
        "R",
        "T",
        "L",
        "L",
        "R",
        "R",
        "L",
        "R",
        "R",
        "R",
        "R",
        "T",
        "R",
        "R",
        "L",
        "R",
        "R",
        "L",
        "L",
        "R",
        "L",
      ],
      "seats": null,
      "tableId": "TopCard000000001",
      "type": "TopCard",
    },
    "logo": "https://cms.casumo.com/wp-content/uploads/2018/06/FootballLogo.png",
    "logoBackground": "https://cms.casumo.com/wp-content/uploads/2018/06/FootballStudio.jpg",
    "name": "Football Studio",
    "slug": "evolution-live-football-studio",
    "tableId": "TopCard000000001",
  },
  Object {
    "disabledForLoggedOut": false,
    "hasPlayForFun": false,
    "inMaintenanceMode": false,
    "jackpotId": null,
    "lobby": Object {
      "betBehind": null,
      "bets": Object {
        "max": 600,
        "min": 5,
        "symbol": "€",
      },
      "image": "https://lob.egcvi.com/thumbnail/pk_uth1_med_L.jpg",
      "players": 2,
      "provider": "evolution",
      "results": null,
      "seats": null,
      "tableId": "UTHTable00000001",
      "type": "UTH",
    },
    "logo": "https://cms.casumo.com/wp-content/uploads/2016/12/Ultimate-logo.png",
    "logoBackground": "https://cms.casumo.com/wp-content/uploads/2016/12/Ultimate-bp.png",
    "name": "Ultimate Texas Hold&#8217;em",
    "slug": "evolution-live-ultimate-texas-holdem",
    "tableId": "UTHTable00000001",
  },
  Object {
    "disabledForLoggedOut": false,
    "hasPlayForFun": false,
    "inMaintenanceMode": false,
    "jackpotId": null,
    "lobby": Object {
      "betBehind": null,
      "bets": Object {
        "max": 5000,
        "min": 0.1,
        "symbol": "€",
      },
      "image": "https://lob.egcvi.com/thumbnail/dc1_mw_med_L.jpg",
      "players": 3,
      "provider": "evolution",
      "results": Array [
        "02",
        "01",
        "01",
        "20",
        "01",
        "X2",
        "05",
        "05",
        "02",
      ],
      "seats": null,
      "tableId": "MOWDream00000001",
      "type": "MoneyWheel",
    },
    "logo": "https://cms.casumo.com/wp-content/uploads/2017/05/dream_catcher_logo.png",
    "logoBackground": "https://cms.casumo.com/wp-content/uploads/2017/09/dream_catcher_bg-copy.jpg",
    "name": "Dream Catcher",
    "slug": "dream-catcher",
    "tableId": "MOWDream00000001",
  },
]
`;
